content/en/user-guide/tools/transparent-endpoint-injection/_index.md

```diff
@@ -3,26 +3,88 @@ title: "Transparent Endpoint Injection"
+no application code changes are required to let your application connect to local cloud APIs because
+Transparent Endpoint Injection uses the integrated [DNS Server]({{< ref "dns-server" >}}) to resolve AWS API calls to target LocalStack.
+
+## Configuration
+
+This section explains the most important configuration options summarized under [Configuration]({{< ref "configuration#dns" >}}).
+
+### Disable transparent endpoint injection
+
+If you do not to use Transparent Endpoint Injection in LocalStack Pro, opt out using:
+
+```bash
+DISABLE_TRANSPARENT_ENDPOINT_INJECTION=1
+```
+
+This option disables DNS resolution of AWS domains to the LocalStack container and prevents Lambda from disabling SSL validation.
+If Transparent Endpoint Injection is _not_ used, the AWS SDK within Lambda functions might connect to the real AWS API.
+Transparent Endpoint Injection is only available in LocalStack Pro.
+
+Alternatively, specific AWS endpoints can be resolved to AWS while continuing to use Transparent Endpoint Injection.
+Refer to the [DNS server configuration]({{< ref "dns-server#configuration" >}}) for skipping selected domain name patterns.
+
+## Self-signed certificates
+
+In LocalStack Pro and Lambda, Transparent Endpoint Injection automatically disables SSL certificate validation of the AWS SDK for the
+most common Lambda runtimes including Python, Node.js, and Java (SDK v1).
+For other services and unsupported Lambda runtimes, you may have to configure your AWS clients to accept self-signed certificates because
+we are repointing AWS domain names (e.g., `*.amazonaws.com`) to `localhost.localstack.cloud`.
+For example, the following command fails with an SSL error:
+
+{{< command >}}
+$ aws kinesis list-streams
+SSL validation failed for https://kinesis.us-east-1.amazonaws.com/ [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1076)
+{{< / command >}}
+
+whereas the following command works:
+
+{{< command >}}
+$ PYTHONWARNINGS=ignore aws --no-verify-ssl kinesis list-streams
+{
+"StreamNames": []
+}
+{{< / command >}}
+
+Disabling SSL validation depends on the programming language and version of the AWS SDK used.
+For example, the [`boto3` AWS SDK for Python](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.client) provides a parameter `verify=False` to disable SSL verification.
+Similar parameters are available for most other [AWS SDKs](https://docs.aws.amazon.com/sdkref/latest/guide/version-support-matrix.html).
+
+For Node.js, you can set this environment variable in your application, to allow the AWS SDK to talk to the local APIs via SSL:
+
+```node.js
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0"
+```
+
+If you are using the Java AWS SDK v2 in Lambda, you can opt in to automatically disable SSL validation using the configuration `LAMBDA_DISABLE_JAVA_SDK_V2_CERTIFICATE_VALIDATION=1`.
```

Comment:
@dfangl @dominikschubert sadly, we missed making this breaking change upon v3 announced at the 2.3.0 release:
> Transparent endpoint injection for Lambdas running AWS Java SDK v2 is now available with the feature flag LAMBDA_DISABLE_JAVA_SDK_V2_CERTIFICATE_VALIDATION=1 . This will be default behaviour by 3.0.
https://discuss.localstack.cloud/t/localstack-release-v2-3-0/533#other-changes-and-enhancements-14

Should we add it to the Lambda configurations (so we won't miss it at v4)?

Reply:
*general config discussion*: do we strive to have all defaults `0` for configurations?

If we ever change the default for a single config, it gets inconsistent. Hence avoiding double negations might be clearer in most cases 💭